DQMServices/FwkIO/plugins/format.h

```diff
@@ -29,6 +29,7 @@ enum TypeIndex {
   kTH2FIndex,
   kTH2SIndex,
   kTH2DIndex,
+  kTH2IIndex,
```

Comment:
Are the integer values of this enumeration stored anywhere persistently? Or is it sufficient to be consistent with `kTypeNames` below?

Reply:
I guess it's a question for @cms-sw/dqm-l2, but these are all the enumeration occurrences  I could find in cmssw: https://github.com/cms-sw/cmssw/search?q=kTH2DIndex&type=

Reply:
Thanks @mmusich, my question was indeed more for @cms-sw/dqm-l2.

If I read the code correctly, the index in this enum gets eventually used in the `DQMRootOutputModule` in
https://github.com/cms-sw/cmssw/blob/ac27d01b6b0bbcefcb2b2971edb5dd06ee92647d/DQMServices/FwkIO/plugins/DQMRootOutputModule.cc#L424-L433
as the `typeIndex`, that then gets stored as `kTypeBranch` branch of `kIndicesTree`.

In `DQMRootSource` this information is then used in
https://github.com/cms-sw/cmssw/blob/ac27d01b6b0bbcefcb2b2971edb5dd06ee92647d/DQMServices/FwkIO/plugins/DQMRootSource.cc#L682-L687

By just reading the code, it seems to me that a `TH3F`-`TTree` in a DQMIO file written before this PR would be interpreted as `TH2I`-`TTree` with this PR. But do we guarantee reading DQMIO files written in older releases?

Reply:
> But do we guarantee reading DQMIO files written in older releases?

speaking for personal experience, it is sometimes convenient to re-run e.g. the harvesting step over older DQMIO files without having to check explicitly the release in which these were produced. Would putting the new element of the enumeration at the end solve the problem here?

Reply:
I'd _think_ that adding the new entry at the end of the enum (i.e. before `kNIndicies`) would solve the backwards compatibility, but I did not trace the code through to be sure about it. Best would be to have a specific tests ensuring that we can properly read old DQMIO files.

Reply:
> Best would be to have specific tests ensuring that we can properly read old DQMIO files

@emanueleusai can you please provide such a test?

Reply:
This is a good idea. I am discussing this with the DQM developers as I don't have enough experience creating unit tests for this case myself.

Reply:
So, I re-ordered the enumeration as suggested in commit https://github.com/cms-sw/cmssw/pull/37665/commits/9a01dcf2a80deabefc73bff8ccdd59be71944aad.
I did run all the available DQM unit tests and all succeeded:

```console
 more ../logs/slc7_amd64_gcc10/testing.log | grep succeeded
---> test TestDQMServicesDemo succeeded
---> test runFastHadd succeeded
---> test testSchemaEvolution succeeded
---> test DQMServicesStreamerIORunStreamer succeeded
---> test TestDQMServicesFwkIOScripts succeeded
---> test TestDataFormatsHistogramsMEtoEDMFormat succeeded
---> test TestDataFormatsHistogramsPath succeeded
```

if there's anything else to test let me know.

Reply:
> if there's anything else to test let me know.

The unit Test TestDQMServicesDemo  as it is not testing your addition of TH2I, this would need the modifications I implemented on files at /afs/cern.ch/user/j/jfernan/public/forMarco    replacing the current versions of DQMServices/Demo/test

This can be done in this PR so that it goes hand by hand, or by @emanueleusai in a separated PR (once this PR is integrated, otherwise the Unit Test will fail evrywhere, of course)

Reply:
>The unit Test TestDQMServicesDemo as it is, it is not testing your addition of TH2I, this would need the modifications I implemented on files at /afs/cern.ch/user/j/jfernan/public/forMarco replacing the current versions of DQMServices/Demo/test
This can be done in this PR so that it goes hand by hand, 

@jfernan2 Thanks Javier! I pushed the updates in commit  [0fcca5b](https://github.com/cms-sw/cmssw/pull/37665/commits/0fcca5b97c81b9a6342156ae478d192c1a8b83fa).
and run the tests, they are still successfull:
```console
$ more ../logs/slc7_amd64_gcc10/testing.log | grep succeeded
---> test TestDQMServicesDemo succeeded
---> test runFastHadd succeeded
---> test testSchemaEvolution succeeded
---> test DQMServicesStreamerIORunStreamer succeeded
---> test TestDQMServicesFwkIOScripts succeeded
---> test TestDataFormatsHistogramsMEtoEDMFormat succeeded
---> test TestDataFormatsHistogramsPath succeeded

```